packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_tradeoptions.js

```diff
@@ -68,6 +69,20 @@ Blockly.Blocks.trade_definition_tradeoptions = {
         };
     },
     onchange(event) {
+        if (event.type === 'change') {
+            const selected_block = this.workspace.getBlockById(event.blockId);
+            selected_block?.parentBlock_?.inputList.filter(
+                item => ['DURATION', 'AMOUNT'].includes(item.name)
+            ).forEach(input => {
+                const input_target = input.connection.targetBlock();
+                const value = input_target.getFieldValue('NUM');
+                if (isFinite(value) && value.startsWith('0')) {
```

Comment:
why do we need to check isFinite() here?

Reply:
When the user clears the input field, the value from `getFieldValue()` becomes `NaN` and we fall in a loop. This check is to prevent that.